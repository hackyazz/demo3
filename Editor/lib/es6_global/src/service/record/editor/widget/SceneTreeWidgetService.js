


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
