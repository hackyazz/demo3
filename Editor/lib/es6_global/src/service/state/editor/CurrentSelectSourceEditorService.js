


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
