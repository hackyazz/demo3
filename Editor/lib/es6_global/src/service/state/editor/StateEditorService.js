

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
