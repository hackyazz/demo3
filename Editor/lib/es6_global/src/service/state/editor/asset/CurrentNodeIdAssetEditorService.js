

import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as CurrentNodeIdAssetService$WonderEditor from "../../../record/editor/asset/CurrentNodeIdAssetService.js";
import * as RootTreeAssetEditorService$WonderEditor from "./RootTreeAssetEditorService.js";

function getCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
}

function unsafeGetCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.unsafeGetCurrentNodeId(editorState[/* assetRecord */2]);
}

function clearCurrentNodeId(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.clearCurrentNodeId(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setCurrentNodeId(currentNodeId, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.setCurrentNodeId(currentNodeId, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function couldRemoveCurrentNode(editorState) {
  var match = CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
  if (match !== undefined) {
    return !NodeAssetService$WonderEditor.isIdEqual(match, NodeAssetService$WonderEditor.getNodeId(RootTreeAssetEditorService$WonderEditor.getRootNode(editorState)));
  } else {
    return false;
  }
}

export {
  getCurrentNodeId ,
  unsafeGetCurrentNodeId ,
  clearCurrentNodeId ,
  setCurrentNodeId ,
  couldRemoveCurrentNode ,
  
}
/* CurrentNodeIdAssetService-WonderEditor Not a pure module */
