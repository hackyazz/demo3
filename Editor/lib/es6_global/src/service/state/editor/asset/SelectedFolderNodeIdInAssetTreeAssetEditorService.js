

import * as SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor from "../../../record/editor/asset/SelectedFolderNodeIdInAssetTreeAssetService.js";

function getSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function unsafeGetSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.unsafeGetSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function clearSelectedFolderNodeIdInAssetTree(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getSelectedFolderNodeIdInAssetTree ,
  unsafeGetSelectedFolderNodeIdInAssetTree ,
  clearSelectedFolderNodeIdInAssetTree ,
  setSelectedFolderNodeIdInAssetTree ,
  
}
/* SelectedFolderNodeIdInAssetTreeAssetService-WonderEditor Not a pure module */
