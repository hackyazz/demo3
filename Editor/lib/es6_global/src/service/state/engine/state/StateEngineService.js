

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
