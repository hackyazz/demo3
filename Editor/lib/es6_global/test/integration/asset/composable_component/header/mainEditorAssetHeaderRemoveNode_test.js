

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTestTool$WonderEditor from "../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../src/service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorAssetTool$WonderEditor from "../../tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as LightMaterialToolEngine$WonderEditor from "../../../../tool/engine/LightMaterialToolEngine.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../tool/MainEditorAssetTreeTool.js";
import * as LightMaterialEngineService$WonderEditor from "../../../../../src/service/state/engine/LightMaterialEngineService.js";
import * as MainEditorLightMaterialTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/renderGroup/material/tool/MainEditorLightMaterialTool.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/CurrentNodeIdAssetEditorService.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../src/service/state/engine/GameObjectComponentEngineService.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../tool/MainEditorAssetHeaderOperateNodeTool.js";
import * as SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/SelectedFolderNodeIdInAssetTreeAssetEditorService.js";

describe("MainEditorAssetHeader->remove node", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                Curry._1(Sinon.restoreSandbox, sandbox[0]);
                StateEditorService$WonderEditor.setState(SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(CurrentNodeIdAssetEditorService$WonderEditor.clearCurrentNodeId(StateEditorService$WonderEditor.getState(/* () */0))));
                return /* () */0;
              }));
        describe("test remove tree node", (function () {
                Wonder_jest.test("if not select specific treeNode, remove-button's disabled props should == true ", (function () {
                        MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
                        MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                      }));
                describe("else", (function () {
                        Wonder_jest.test("remove-button's disabled props should == false", (function () {
                                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
                                var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData), undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                              }));
                        describe("test select folder", (function () {
                                beforeEach((function () {
                                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
                                      }));
                                return Wonder_jest.test("click remove-button should remove folder from assetTreeRoot", (function () {
                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                              BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0);
                                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeFolderNode(undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData), /* () */0);
                                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                                            }));
                              }));
                        describe("test select file", (function () {
                                beforeEach((function () {
                                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorAssetTool$WonderEditor.initAssetTree);
                                      }));
                                describe("test texture", (function () {
                                        describe("select texture;\n            click remove-button;\n            ", (function () {
                                                return Wonder_jest.test("should remove it from assetTreeRoot", (function () {
                                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildOneTextureAssetTree */0], /* () */0);
                                                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeTextureNode(undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData), /* () */0);
                                                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                                                            }));
                                              }));
                                        describe("select texture;\n            drag texture to set gameObject material map;\n            select texture;\n            click remove-button;\n            ", (function () {
                                                describe("should remove it from engineState", (function () {
                                                        beforeEach((function () {
                                                                var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
                                                                StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                                                                        return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                                                                      }));
                                                                return MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                                                              }));
                                                        describe("should remove it from scene->materials", (function () {
                                                                describe("test remove lightMaterial->diffuseMap", (function () {
                                                                        var _drag = function (assetTreeData) {
                                                                          return MainEditorLightMaterialTool$WonderEditor.Drag[/* dragAssetTextureToMap */0](undefined, undefined, undefined, undefined, undefined, undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData), /* () */0);
                                                                        };
                                                                        var _remove = function (assetTreeData) {
                                                                          return MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeTextureNode(undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* getFirstTextureNodeId */2], assetTreeData), /* () */0);
                                                                        };
                                                                        Wonder_jest.test("test one gameObject use one material", (function () {
                                                                                var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildOneTextureAssetTree */0], /* () */0);
                                                                                _drag(assetTreeData);
                                                                                _remove(assetTreeData);
                                                                                MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                                                                                var lightMaterial = GameObjectTool$WonderEditor.getCurrentGameObjectLightMaterial(/* () */0);
                                                                                var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                                                                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](LightMaterialEngineService$WonderEditor.getLightMaterialDiffuseMap(lightMaterial, engineState)), undefined);
                                                                              }));
                                                                        describe("test two gameObjects use one material", (function () {
                                                                                return Wonder_jest.test("test gameObjects are in scene", (function () {
                                                                                              var currentGameObject = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                                                                              var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                                                                              var oldMaterial = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(currentGameObject, engineState);
                                                                                              var secondCubeGameObject = MainEditorSceneTool$WonderEditor.getCubeByIndex(1, engineState);
                                                                                              var engineState$1 = LightMaterialToolEngine$WonderEditor.replaceGameObjectLightMaterial(secondCubeGameObject, oldMaterial, engineState);
                                                                                              StateEngineService$WonderEditor.setState(engineState$1);
                                                                                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Texture */2][/* buildOneTextureAssetTree */0], /* () */0);
                                                                                              _drag(assetTreeData);
                                                                                              MainEditorSceneTool$WonderEditor.setSecondCubeToBeCurrentSceneTreeNode(/* () */0);
                                                                                              _drag(assetTreeData);
                                                                                              _remove(assetTreeData);
                                                                                              var engineState$2 = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                                                                              var newMaterial1 = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(currentGameObject, engineState$2);
                                                                                              var newMaterial2 = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(secondCubeGameObject, engineState$2);
                                                                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                                                                              LightMaterialEngineService$WonderEditor.getLightMaterialDiffuseMap(newMaterial1, engineState$2),
                                                                                                              LightMaterialEngineService$WonderEditor.getLightMaterialDiffuseMap(newMaterial2, engineState$2)
                                                                                                            ]), /* tuple */[
                                                                                                          undefined,
                                                                                                          undefined
                                                                                                        ]);
                                                                                            }));
                                                                              }));
                                                                        return /* () */0;
                                                                      }));
                                                                return /* () */0;
                                                              }));
                                                        return /* () */0;
                                                      }));
                                                return /* () */0;
                                              }));
                                        return /* () */0;
                                      }));
                                return /* () */0;
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
