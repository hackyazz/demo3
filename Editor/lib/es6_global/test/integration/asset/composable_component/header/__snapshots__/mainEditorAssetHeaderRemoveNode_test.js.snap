// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->remove node test remove tree node else remove-button's disabled props should == false 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/remove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <div
              className="draggable-container select-active"
              draggable={true}
              onClick={[Function]}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              NoName Folder
            </div>
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  />
</article>
`;

exports[`MainEditorAssetHeader->remove node test remove tree node else test select file test texture select texture;
            click remove-button;
             should remove it from assetTreeRoot 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  />
</article>
`;

exports[`MainEditorAssetHeader->remove node test remove tree node else test select folder click remove-button should remove folder from assetTreeRoot 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  />
</article>
`;

exports[`MainEditorAssetHeader->remove node test remove tree node if not select specific treeNode, remove-button's disabled props should == true  1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  />
</article>
`;
